python/cudf/dataframe/index.py

```diff
@@ -205,7 +206,14 @@ def __ge__(self, other):
     def equals(self, other):
         if len(self) != len(other):
             return False
-        return (self == other)._values.all()
+        elif len(self) == 1:
+            return self[0] == other[0]
```

Comment:
Was this necessary or just an optimization?

Reply:
Haha no I did not refactor this idly. It is ugly and I hate it, but I don't have a simper factoring. The danger has to do with something like `self == other` evaluating all the way down to a boolean, thus breaking `_values`. Each step was necessary for reasons I can't directly recall.